test/e2e/infra/secondary_network/prepare_cluster.sh

```diff
@@ -0,0 +1,358 @@
+	fi
+	echo " $action VirtualNetworks Done!"
+}
+
+function WhereaboutsCNI() {
+	kubectl $action -f $SEC_NET_REMOTE_CONFIG_DIR_PATH$YAML_WHEREABOUTS_IP_POOL
+	if [ $? -ne 0 ]; then
+		echoerr "Failed to $action Whereabouts CNI"
+		exit 1
+	fi
+	echo "$action WhereaboutsCNI Done!"
+
+	kubectl $action -f $SEC_NET_REMOTE_CONFIG_DIR_PATH$YAML_WHEREABOUTS_OVERLAPPING_IP_RANGE_RES
+	if [ $? -ne 0 ]; then
+		echoerr "Failed to $action Whereabouts overlapping IP Range reservation"
+		exit 1
+	fi
+	echo "$action WhereaboutsOverlappingIpRangeRes Done!"
+}
+
+function AntreaCNI() {
+	kubectl $action -f $SEC_NET_REMOTE_CONFIG_DIR_PATH$YAML_ANTREA -n $NAMESPACE
+        if [ $? -ne 0 ]; then
+              echoerr "Failed to $action Antrea CNI"
+              exit 1
+        fi
+	IFS=$'\n' read -r -d '' -a antrea_agents < <(kubectl get pods -n $NAMESPACE --no-headers -o custom-columns=":metadata.name" | grep antrea-agent && printf '\0')
+	for agent in ${antrea_agents[@]};do
+		iterate=0
+		while [ $iterate -ne 30 ]; do
+			if [[ $(kubectl get pods -n $NAMESPACE $agent -o 'jsonpath={..status.conditions[?(@.type=="Ready")].status}') != "True" ]]
+			then
+				echoerr "waiting for $agent to be ready."
+				sleep 1
+			else
+				echoerr "$agent is ready"
+				break
+			fi
+			iterate=$((iterate+1))
+		done
+	done
+        if [ $iterate -eq 30 ]; then
+              echoerr "$agent Antrea CNI deployment failed or timedout"
+              exit 1
+        fi
+}
+
+function generateAntreaConfig() {
+	genManifest=$THIS_DIR/../../../../hack/generate-manifest.sh
+	$genManifest --sriov --whereabouts > "$SEC_NET_LOCAL_CONFIG_DIR_PATH"antrea.yml
+        if [ $? -ne 0 ]; then
+              echoerr "Failed to generate Antrea.yml"
+              exit 1
+        fi
+	echo "Antrea.yml generated with whereabouts and Intel sriov configuration!"
+}
+
+function scpConfigFilesToTargetNode() {
+	ssh $hostIP "rm -rf $SEC_NET_REMOTE_CONFIG_DIR_PATH"
+	scp -r $SEC_NET_LOCAL_CONFIG_DIR_PATH $hostIP:/tmp
+	if [ $? -ne 0 ]; then
+		echoerr "Failed to copy the downloaded configuration files!"
+		exit 1
+	fi
+	rm -rf $SEC_NET_LOCAL_CONFIG_DIR_PATH
+	echo "Downloaded Configuration files are copied successfully! "
+}
+
+function configureSecondaryNetworkPrerequisite() {
+        ssh vcesd2 <<EOF
+        export interface=$interface vfs_num=$vfs_num SEC_NET_REMOTE_CONFIG_DIR_PATH=$SEC_NET_REMOTE_CONFIG_DIR_PATH YAML_CONFIG_MAP=$YAML_CONFIG_MAP NAMESPACE=$NAMESPACE  net_instance=$net_instance action=$action
+        if [ ! -z $interface ] && [ ! -z $vfs_num ]
+        then
+           $(typeset -f echoerr checkInterface setIPLinkDown configureVFs setIPLinkUp SriovDevicePlugin)
+           setIPLinkDown
+           configureVFs
+           setIPLinkUp
+           SriovDevicePlugin
+        fi
+        $(typeset -f echoerr NetworkAttachmentDefinition VirtualNetworks WhereaboutsCNI AntreaCNI)
+        NetworkAttachmentDefinition
+        VirtualNetworks
+        WhereaboutsCNI
+        AntreaCNI
```

Comment:
all these commands are run over SSH and they all use kubectl
was it a conscious decision, and if yes why? could we instead assume that kubectl is available locally and avoid running these commands over SSH? IMO, it's more likely that kubectl + kubeconfig are available locally, and it's better than requiring that they be present on the K8s Node

Reply:
ok, whatever works best for CI
we will probably not use this script for testing with veth pairs though: we run CI e2e tests with Kind whenever we can, and we don't have SSH access to the Nodes. Vagrant is only used for local development as it can be more convenient that Kind in some cases.

Reply:
ah ok. So, should we redo this to use kubeconfig instead of ssh?. so, that the same script can be used for both the cases?. for Intel server/SRIOV case, we can add additional step to SCP the /etc/kubernetes/admin.conf from intel server to VMware test bad custom location, prior to the secondary network setup and do handle everything with kubectl command locally?.

Reply:
if you want the script to be usable for both, then I would say yes
I am not sure it will translate to "an extra step" though: yes, there will be an SCP step to copy the kubeconfig file from the server to the local machine, but it also means that we can remove some of the existing stuff (download of YAMLs + SCP to server).

I wouldn't be surprised if in the future we break this script into 2 parts: SRIOV configuration of _a_ K8s Nodde (requires SSH) / configuration of the cluster (only requires kubectl + kubeconfig).

Reply:
I have updated the script as suggested. 
But, with little more thought, I do think its better to split this script into 2 parts (1 for SRIOV and 1 for veth - probably split the common function calls into separate file and source/use it at SRIOV and veth pair specific execs).
For now, we use ssh for SRIOV related function calls (i'e to config VFs, interface up/down etc.,). scp /etc/kubernetes/admin.conf from remote node to the local and export KUBECONFIG to the custom located admin.conf and use kubectl to apply the URLs/ and the locally built yml files.